x-pack/plugins/watcher/public/sections/watch_edit/components/threshold_watch_edit/threshold_watch_action_dropdown.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  EuiSpacer,
+  EuiSuperSelect,
+  EuiText,
+  EuiFlexItem,
+  EuiIcon,
+  EuiFlexGroup,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useContext } from 'react';
+import { Action } from 'plugins/watcher/models/action';
+import { ACTION_TYPES } from '../../../../../common/constants';
+import { WatchContext } from '../../watch_context';
+
+const EMPTY_FIRST_OPTION_VALUE = 'empty-first-option';
+
+const disabledMessage = i18n.translate(
+  'xpack.watcher.sections.watchEdit.actions.disabledOptionLabel',
+  {
+    defaultMessage: 'Disabled. Configure elasticsearch.yml.',
+  }
+);
+
+const firstActionOption = {
+  inputDisplay: i18n.translate('xpack.watcher.sections.watchEdit.actions.emptyFirstOptionLabel', {
+    defaultMessage: 'Add an action',
+  }),
+  value: EMPTY_FIRST_OPTION_VALUE,
+};
+
+interface Props {
+  settings: {
+    actionTypes: {
+      [key: string]: {
+        enabled: boolean;
+      };
+    };
+  } | null;
+}
+
+export const WatchActionsDropdown: React.FunctionComponent<Props> = ({ settings }) => {
+  const { addAction } = useContext(WatchContext);
+
+  const allActionTypes = Action.getActionTypes();
+
+  const actions = Object.keys(allActionTypes).map(actionKey => {
+    const { typeName, iconClass, selectMessage } = allActionTypes[actionKey];
```

Comment:
nit: you could simplify with `Object.entries`

```js
const allActionTypes = Action.getActionTypes() as Record<string, any>;

const actions = Object.entries(allActionTypes).map(([type, { typeName, iconClass, selectMessage }]) => ({
  type,
  typeName,
  iconClass,
  selectMessage,
  isEnabled: settings ? settings.actionTypes[type].enabled : true,
}));
```

Reply:
thanks - updated